### DISCUSSION TITLE
adding a `pool` feature without any specific pool crate enabled

### DISCUSSION THREAD

**Discussion by fakeshadow** [weiznich/diesel_async#187] [Ideas]
I'm trying to implement my own db driver and pool implement as an extension crate of `diesel-async`. In the progress I find `pooled_connection` module can only be enabled with at least one existing pool crate. Resulting extra dependencies I don't need.

I imagine a feature like this would help resolve the dependency issue:
`Cargo.toml`
```toml
[features]
pool = []
bb8 = ["pool", "dep::bb8"]
```

`mod.rs`
```rust
[cfg(feature = "pool")]
pub mod pooled_connection;
```

**Comment by weiznich** [comment]
That sounds reasonable, happy to accept a PR for this.
